ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestSuite.scala

```diff
@@ -37,21 +37,28 @@ private[testtool] abstract class LedgerTestSuite {
     )((ec: ExecutionContext) => (_: Seq[ParticipantTestContext]) => testCase(ec))
   }
 
+  /** @param cleanupNewUsers Keeps track of users present before the test case started and after it has finished and deletes difference.
+    *                            NOTE: If set then the test case will not run concurrently.
+    */
   protected final def testGivenAllParticipants(
       shortIdentifier: String,
       description: String,
-      participants: ParticipantAllocation,
+      partyAllocation: PartyAllocation,
       timeoutScale: Double = 1.0,
       runConcurrently: Boolean = true,
       repeated: Int = 1,
       enabled: Features => Boolean = _ => true,
       disabledReason: String = "No reason",
+      cleanupNewUsers: Boolean = false,
   )(
       testCase: ExecutionContext => Seq[ParticipantTestContext] => PartialFunction[
         Participants,
         Future[Unit],
       ]
   ): Unit = {
+    if (cleanupNewUsers) {
+      assert(!runConcurrently, "Cleaning-up of created user cannot be run concurrently!")
```

Comment:
This looks like a smell. I'm not in touch with the test tool anymore, but we used to use parties to isolate tests. Can you isolate user tests in the same way, by having per-test-case users?

Reply:
This code is a left-over I failed to delete from the PR. 
The up-to-date approach is to have a wrapper around `userManagement.createUser` that keeps track what users where created for each test-case

Reply:
Deleted